cypress/utils/utils.ts

```diff
@@ -668,7 +668,43 @@ export function verifyImportErrorMsg(errorMsg: any): void {
     }
 }
 
+export function migration_wave_kebab_menu(menu): void {
+    // The value for menu could be one of {Export to Issue Manager, Delete}
+    cy.get(actionButton).eq(1).click({ force: true });
+    cy.get(commonView.kebabMenuItem).contains(menu).click({ force: true });
+}
+
+export function deleteAllMigrationWaves(currentPage = false): void {
+    MigrationWave.open();
+    cy.wait(2000);
+    cy.get(waveTable)
+        .next()
+        .then(($div) => {
+            if (!$div.hasClass("pf-c-empty-state")) {
+                cy.wait(1000);
+                cy.get("span.pf-c-options-menu__toggle-text")
+                    .eq(0)
+                    .then(($body) => {
+                        if (!$body.text().includes("of 0")) {
+                            if (currentPage) {
+                                cy.get(".pf-c-dropdown__toggle-button").click({ force: true });
+                                clickByText(button, "Select page");
+                            } else {
+                                cy.get("input#bulk-selected-items-checkbox", {
+                                    timeout: 10 * SEC,
+                                }).check({ force: true });
+                            }
+
+                            migration_wave_kebab_menu("Delete");
+                            clickByText(button, "Delete", true);
+                        }
+                    });
+            }
+        });
+}
+
 export function deleteApplicationTableRows(currentPage = false): void {
+    deleteAllMigrationWaves();
```

Comment:
DeleteallApplications is used at  many places .
and now we have to delete all migration waves too before other tests . I think it is not a good design in long term as we have seen while using cypress parallel or cypress split . 

Tests should cleanup entities created in the test in the after method. 

Let's try to reduce usage of deleteAllApplications () and that method should just delete applications not waves .

Reply:
Ideally, the tests that create migration waves should also delete migration waves. But, if for some reason, the migration waves don't get deleted, we end up in this situation . That's why the need for this function.

Reply:
I have modified the deleteApplicationTable() method to not fail if Delete method is disabled and we will start removing it from tests . so you can remove this line from here.

Reply:
Done